From 1dd1e763873c2e19ec9cfc8134b91406c36cac30 Mon Sep 17 00:00:00 2001
From: Matthias Reichl <hias@horus.com>
Date: Sat, 10 Oct 2015 12:29:18 +0200
Subject: [PATCH] bcm2835-dma: Fix dreq not set for slave transfers

Set dreq to slave_id if it is not set like in bcm2708-dmaengine.
---
 drivers/dma/bcm2835-dma.c | 2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/dma/bcm2835-dma.c
+++ b/drivers/dma/bcm2835-dma.c
@@ -657,6 +657,8 @@ static int bcm2835_dma_slave_config(stru
 	}
 
 	c->cfg = *cfg;
+	if (!c->dreq)
+		c->dreq = cfg->slave_id;
 
 	return 0;
 }
